xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3644,9 +3642,17 @@ void CVideoPlayer::SetSpeed(float speed)
 
   if (!CanSeek())
     return;
-  
+  float m_oldPlayerSpeed = m_playSpeed;
   m_newPlaySpeed = speed * DVD_PLAYSPEED_NORMAL;
   SetPlaySpeed(speed * DVD_PLAYSPEED_NORMAL);
+  if (m_oldPlayerSpeed == 0)
```

Comment:
Remember what I said on the other PR: player commands are asynchronous. Means it does not matter if you call the callback before or after SetPlaySpeed, speed won't have changed anyway at this time. No need for m_oldPlayerSpeed, compare m_newPlaySpeed with m_PlaySpeed.
The behaviour is as follows:
resume: this is whenever new speed is normal and old speed was not normal speed
pause: new speed is DVD_PLAYSPEED_PAUSE and old speed was not DVD_PLAYSPEED_PAUSE